js/glowingbear.js

```diff
@@ -746,6 +746,31 @@ weechat.controller('WeechatCtrl', ['$rootScope', '$scope', '$store', '$timeout',
         }
     };
 
+    function nickFromBufferline(bufferline) {
+        if (!bufferline || !bufferline.prefix) {
+            return '';
+        }
+        var prefix = bufferline.prefix;
+        var nick="";
+        for (var i=0; i < prefix.length; i++) {
+            nick += prefix[i].text;
+        }
+        return nick;
+    }
+    $rootScope.bufferlineClassifier = function(lines, index) {
```

Comment:
why is this incredibly inefficient? it's passed by reference, isn't it? I don't iterate over the entire lines[] array